### PR TITLE
feat: cast List / LargeList to Utf8 / LargeUtf8

### PR DESCRIPTION
**Pull request by gandronchik**
cast List / LargeList to Utf8 / LargeUtf8

### CODE REVIEW

File: arrow/src/util/display.rs [hunk 1]
```diff
@@ -235,6 +235,22 @@ macro_rules! make_string_from_list {
...
+    ($column: ident, $row: ident) => {{
+        let list = $column
+            .as_any()
+            .downcast_ref::<array::LargeListArray>()
+            .ok_or(ArrowError::InvalidArgumentError(format!(
+                "Repl error: could not convert large list column to list array."
+            )))?
+            .value($row);
+        let string_values = (0..list.len())
+            .map(|i| array_value_to_string(&list.clone(), i))
```
Comment (tustvold) [inline arrow/src/util/display.rs:248]: Why the clone?

**Review by tustvold — APPROVED** [review]

**Comment by alamb** [comment]
@gandronchik  perhaps you can respond to @tustvold 's question? Then we can merge the PR?

**Comment by gandronchik** [comment]
> @gandronchik perhaps you can respond to @tustvold 's question? Then we can merge the PR?

Sorry for delay, I will do it in next few days.

**Comment by alamb** [comment]
Ping -- how are we doing on getting this PR merged?

**Comment by gandronchik** [comment]
> Ping -- how are we doing on getting this PR merged?

@alamb sorry for the delay. Now it it ready to merge

**Comment by alamb** [comment]
Thanks @gandronchik  and @tustvold

**Pull request merged**